src/iexamine.cpp

```diff
@@ -125,6 +125,61 @@ void iexamine::cvdmachine( player &p, const tripoint & ) {
     p.mod_moves( -1000 );
 }
 
+/**
+ * UI FOR LAB_FINALE NANO FABRICATOR.
+ */
+void iexamine::nanofab( player &p, const tripoint &examp )
+{
+    bool table_exists = false;
+    tripoint spawn_point;
+    for( const auto &valid_location : g->m.points_in_radius( examp, 1, 0 ) ) {
+        if( g->m.ter(valid_location) == t_nanofab_body ) {
+            spawn_point = valid_location;
+            table_exists = true;
+            break;
+        }
+    }
+    if (!table_exists){
+        return;
+    }
+
+    auto nanofab_template= g->inv_map_splice( []( const item &e ) {
+        return ( e.has_var( "NANOFAB_ITEM_ID" ) );
+    }, _( "Introduce Nanofabricator template" ), PICKUP_RANGE, _( "You don't have any usable templates." ) );
+
+    if( !nanofab_template ) {
+        return;
+    }
+
+    //item *it = nanofab_template.get_item();
+    item new_item( nanofab_template->get_var( "NANOFAB_ITEM_ID" ), calendar::turn );
+
+    auto qty = new_item.volume() / units::legacy_volume_factor;
+    qty = std::max( 1, qty );
+    auto reqs = *requirement_id( "nanofabricator" ) * qty;
+
+    if( !reqs.can_make_with_inventory( p.crafting_inventory() ) ) {
+        popup( "%s", reqs.list_missing().c_str() );
```

Comment:
Will this line get translated or not?

Reply:
Yes it will, the missing requirement text is handled in  requirement_data::list_missing() the other stuff is just the name for the nanomaterial canister as defined in its json object, which is translated separtedly.